File: frontend/src/services/interview.ts
import api from "./api";

export interface Interview {
  id: number;
  user_id: number;
  title: string;
  role: string;
  company?: string | null;
  difficulty: string;
  interview_type: string;
  job_description?: string | null;
  status: string;
  created_at: string;
  updated_at?: string | null;
  resume_id?: number | null;
}

export interface CreateInterviewData {
  title: string;
  role: string;
  company?: string;
  difficulty: string;
  interview_type: string;
  job_description?: string;
  resume_id?: number | null;
}

export async function createInterview(
  data: CreateInterviewData
): Promise<Interview> {
  const response = await api.post("/api/interviews", data);
  return response.data;
}

export async function getInterviews(): Promise<Interview[]> {
  const response = await api.get("/api/interviews");
  return response.data;
}

export async function getInterview(id: number): Promise<Interview> {
  const response = await api.get(`/api/interviews/${id}`);
  return response.data;
}

export async function deleteInterview(id: number) {
  const response = await api.delete(`/api/interviews/${id}`);
  return response.data;
}

File: backend/app/schemas/interview.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class InterviewCreate(BaseModel):
    title: str
    role: str
    company: Optional[str] = None
    difficulty: str
    interview_type: str
    job_description: Optional[str] = None
    resume_id: Optional[int] = None

class InterviewResponse(BaseModel):
    id: int
    user_id: int
    title: str
    role: str
    company: Optional[str]
    difficulty: str
    interview_type: str
    job_description: Optional[str]
    status: str
    created_at: datetime
    updated_at: Optional[datetime]
    resume_id: Optional[int]
    class Config:
        from_attributes = True